CONTRIBUTING.md

```diff
@@ -89,6 +89,43 @@ syntax.
   - Does it only contain single line comments? Or only multi-line comments?
   - Is just C style comments? `/* */, //`
 
+Some languages have a single, standard filename, like Makefile or Dockerfile.
+These can be defined with the `filenames` property:
+
+```json
+"Makefile":{
+    "filenames":[
+        "makefile"
+    ],
+    "extensions":[
+        "makefile",
+        "mak",
+        "mk"
+    ]
+}
+```
+
+Filenames should be all-lowercase, whether or not the filename typically has capital letters included.
+
+Note that filenames will *override* extensions, so with the following
+configuration:
+
+```json
+"Toml":{
+    "extensions": [
+        "toml"
+    ]
+},
+"Cargo":{
```

Comment:
This a bad example, as if someone made a PR with Cargo I would reject it. Consider an example using `Rakefile`, or `Makefile